Main.js

```diff
@@ -1,24 +1,60 @@
-import React from 'react'
-import { StatusBar } from 'expo-status-bar'
-import { View, StyleSheet } from 'react-native'
+import React, { useState } from 'react'
+import { NavigationContainer } from '@react-navigation/native'
+import { createStackNavigator } from '@react-navigation/stack'
 
+import { upsert } from './lib/secrets'
 import { useAuthenticationContext } from './context/authentication'
 import Auth from './components/Auth'
 import Home from './components/Home'
+import TypeNewSecret from './components/TypeNewSecret'
+
+const Stack = createStackNavigator()
+
+const UI_STRINGS = {
+  routes: {
+    home: {
+      title: 'Optic',
+    },
+    type: {
+      title: 'New Secret',
+    },
+  },
+}
 
 export default function Main() {
+  const [secrets, setSecrets] = useState([])
   const { user } = useAuthenticationContext()
+
+  const addSecret = async secret => {
+    const uid = user.uid
+    await upsert({ uid, ...secret })
+    setSecrets(await fetch({ uid }))
+  }
+
   return (
-    <View style={styles.container}>
-      {user ? <Home user={user} /> : <Auth />}
-      <StatusBar style="auto" />
-    </View>
+    <NavigationContainer>
+      {user ? (
+        <Stack.Navigator initialRouteName="Home">
+          <Stack.Screen
+            name="Home"
+            options={{ title: UI_STRINGS.routes.home.title }}
+          >
+            {props => <Home {...props} user={user} secrets={secrets} />}
+          </Stack.Screen>
+          <Stack.Screen
+            name="Type"
+            options={{ title: UI_STRINGS.routes.type.title }}
+          >
+            {props => (
+              <TypeNewSecret {...props} user={user} addSecret={addSecret} />
```

Comment:
the user object can be grabbed from React context, let's not pass it around as props so we can simplify the way we declare these screens. We can do without passing any props to the components